test/gtest/uct/test_sockaddr.cc

```diff
@@ -1043,6 +1048,76 @@ UCS_TEST_P(test_uct_sockaddr, ep_disconnect_err_codes)
 UCT_INSTANTIATE_SOCKADDR_TEST_CASE(test_uct_sockaddr)
 
 
+class test_uct_sockaddr_listener : public test_uct_sockaddr {
+protected:
+    static void conn_request_cb(
+            uct_listener_h listener, void *arg,
+            const uct_cm_listener_conn_request_args_t *conn_req_args)
+    {
+        test_uct_sockaddr_listener *self =
+                static_cast<test_uct_sockaddr_listener*>(arg);
+
+        ASSERT_FALSE(self->m_listener_closed);
+
+        self->handle_conn_request(listener, conn_req_args,
+                                  test_uct_sockaddr::server_connect_cb,
+                                  test_uct_sockaddr::server_disconnect_cb);
+
+        // Wait until listener will be set for being destroyed
+        while (self->m_listener_to_close == nullptr) {
+            std::this_thread::yield();
+        }
+
+        // Wait for 0.5s to give a chance for destroying the listener
+        std::this_thread::sleep_for(std::chrono::milliseconds(500));
```

Comment:
Helps only allocating large piece of memory, which takes >= than 0.5s.
Just doing yield several times doesn’t help.
So, I decided to use sleep(), because it has predictable time limit and 0.5s gives good reproducibility. It may be not reproduced, but in my experiments, it was reproduced every time

Reply:
It seems weird to me, uct_listener_destroy can't take 0.5s. maybe do smth like below?
```
for (int i = 0; i < 10; ++i) {
    std::vector<uint8_t> v(i*page_size, 0);
    std::this_thread::yield();
    std::fill(v.begin(), v.end(). 1);
}
```

Reply:
> It seems weird to me, uct_listener_destroy can't take 0.5s

It is approximately.

> maybe do smth like below?

I think it is better to do `sleep()`. It is more obvious

Reply:
I'm afraid sleep for 0.5s will significantly increase regular CI testing time

Reply:
Running 27 tests took 13.6 seconds on `swx-ucx`: `27 tests from tcp/test_uct_sockaddr_listener (13638 ms total)`
I don't think that it is critical.

Reply:
Decreased time from 500ms to 100ms and it also works fine, so now we get:
`27 tests from tcp/test_uct_sockaddr_listener (2833 ms total)`

Reply:
I still disagree with sleep in tests but let's go with it for now and ask other reviewers.

Reply:
> I still disagree with sleep in tests but let's go with it for now and ask other reviewers.

ok, let's see what other reviewers will propose

Reply:
Noone like sleeps in test but I cannot imagine way to make this test deterministic except patching mutexes implementation.
But I think this way doesn't worth implementation now. Agree to leave it as is for now.